autofaiss/external/distributed.py

```diff
@@ -0,0 +1,313 @@
+import math
+import multiprocessing
+import os
+import shutil
+from itertools import repeat
+from multiprocessing.pool import ThreadPool
+from tempfile import NamedTemporaryFile
+from typing import List, Optional, Union, Iterator, Tuple
+
+import faiss
+import fsspec
+import numpy as np
+from tqdm import tqdm
+
+from autofaiss.readers.embeddings_iterators import get_file_list, ParquetLazyNdArray, ParquetMatrixReader
+from autofaiss.utils.decorators import Timeit
+
+
+def _yield_embeddings_batch(
+    embeddings_paths: List[str],
+    chunk_sizes: List[int],
+    start: int,
+    end: int,
+    embedding_column_name: str
+):
+    """Lazy load a batch of embeddings."""
+    cur_start = cur_end = 0
+    for chunk_size, file_path in zip(chunk_sizes, embeddings_paths):
+        cur_end += chunk_size
+        if cur_end < start:
+            cur_start += chunk_size
+            continue
+        parquet_array = ParquetLazyNdArray(f=file_path, embedding_column_name=embedding_column_name)
```

Comment:
The problem of not using read embeddings is you're losing the features it supports:
* npy files
* reading embedding keys in parquet files

Let's think if it could be possible to use it / to refactor read embeddings so it works for both the distributed and the non distributed case

Reply:
In my last commit, I added the support for npy files and embeddings Ids.
I reused the function get_matrix_reader, but I didn't find a good way to refactor read_embeddings.